document/CoreFormat.mdk

```diff
@@ -413,6 +413,44 @@ The `results` array shall be empty if the run that produced the `run` object did
 If the tool was run solely for the purpose of exporting rule metadata (see &sect;[#run-rules]),
 the `results` property shall be absent.
 
+### `toolNotifications` { #run-toolNotifications }
+
+A `run` object may contain a property named `toolNotifications`
+whose value is an array of zero or more `notification` objects (&sect;[#notification]).
+Each element of the array represents a runtime condition detected by the tool.
+The presence within this array of any `notification` object whose `level` property (&sect;[#notification-level]) is `error` shall mean
+that the run failed.
+
+~ IsoNote
+Depending on the nature of the error, a tool that encounters a runtime error might or might not be able to continue running.
+
+If the error occurs in the course of evaluating a rule, the tool might report the error in `toolNotifications`, disable the rule,
+and continue to execute the remaining rules.
+
+If the error occurs outside of the evaluation of a rule, the tool might report
+the error in `toolNotifications` and then halt. The tool might also just exit, either normally or abnormally.
+~
+
+### `configurationNotifications` { #run-configurationNotifications }
+
+A `run` object may contain a property named `configurationNotifications`
+whose value is an array of zero or more `notification` objects (&sect;[#notification]).
+Each element of the array represents a condition relevant to the tool's configuration.
+The presence within this array of any `notification` object whose `level` property (&sect;[#notification-level]) is `error` shall mean
+that the run failed.
+
+~ IsoNote
+Many tools can be parameterized with information about which rules to run, and how they should be configured.
+In some cases, if the configuration information is invalid, the tool can ignore the invalid information
+and continue to run.
+~
+
+~ IsoExample
+Suppose a tool is invoked with a configuration file which specifies that the tool should disable rule X.
+If there is no such rule as rule X, then the tool should report it in `configurationNotifications`. The tool might
+continue to run, reporting results for the rules that are correctly configured.
```

Comment:
What about a bad command-line argument? An output file path where a file already exists or that is inaccessible? I feel like we could use more examples (i.e., ones that are errors and block execution). I like your example for a configuration warning, where analysis clearly could continue (and probably should)